### ISSUE TITLE
[XItemStack] - serialized Material output of item is incorrect

### ISSUE THREAD

**Issue by MayBlock** [CryptoMorin/XSeries#209]
![avatar](https://i.328888.xyz/2023/04/16/iEsUp5.png)
```kotlin
val config = MemoryConfiguration().also { config ->
    XItemStack.serialize(ItemStack(Material.CLAY_BRICK), config)
} as ConfigurationSection
val item = XItemStack.deserialize(config)
sender.inventory.addItem(item)
sender.sendMessage("you get item ${item.type.name}")
```
Material CLAY_BRICK will be incorrectly serialized as BRICK, plz fix 😘

Server Version: Spigot 1.8.8

**Comment by MayBlock** [comment]
```kotlin
val clayBrick1 = XMaterial.matchXMaterial(Material.CLAY_BRICK).parseItem()
val clayBrick2 = XMaterial.matchXMaterial(ItemStack(Material.CLAY_BRICK)).parseItem()
val clayBrick3 = ItemStack(XMaterial.matchXMaterial(Material.CLAY_BRICK).parseMaterial()!!)
sender.inventory.addItem(clayBrick1, clayBrick2, clayBrick3)
```
these codes are working fine, it seem that the problem only happens with serialize

**Comment by CryptoMorin** [comment]
You didn't show your item() function definition.

**Comment by MayBlock** [comment]
> ~You didn't show your item() function definition.~
> 
> Oh, nvm. It seems like you posted the same code in both screenshot and text forms, I'm not sure why, but it doesn't matter.
> 
> I'll wait for someone else to also confirm this issue.

you can copy this code to reproduce bug

**Comment by MayBlock** [comment]
after my debug, XItemStack is good, i found problem is with matchDefinedXMaterial function in XMaterial class
![bug](https://user-images.githubusercontent.com/49287036/232765782-072e8585-b98f-4ba8-bb00-81efb721ff25.png)
in this function, input name is BRICK, which is get from serialized name of Material CLAY_BRICK by XItemStack, and in this function, it mistakenly recognizes BRICK to BRICKS. result in incorrect output
![bug1](https://user-images.githubusercontent.com/49287036/232767268-58b6a0ec-4204-491e-b8f8-dbbc359daec6.png)
i not sure if this problem occurs with other Material, but in CLAY_BRICK, it does happen

so my current workaround is
```kotlin
private object XMaterialReflection {

    val X_MATERIAL_CLASS by lazy { XMaterial::class.java }
    val X_MATERIAL_METHOD by lazy { X_MATERIAL_CLASS.getDeclaredMethod("matchDefinedXMaterial", String::class.java, Byte::class.java).also { it.isAccessible = true } }
}

fun matchXMaterial(name: String, data: Byte = -1): XMaterial? {
    return runCatching {
        XMaterial.valueOf(name)
    }.getOrNull() ?: run {
        X_MATERIAL_METHOD.invoke(null, name, data) as? Optional<XMaterial>
    }?.get()
}
```
since i haven't delved more into this project, i not sure if this workaround will affect other Material or ItemStack, but at least i haven't encountered problem so far

**Comment by CryptoMorin** [comment]
Fixed in v9.4.0
